Applications/AdminUi/apps/admin_ui/lib/screens/splash_screen.dart

```diff
@@ -45,18 +45,23 @@ class _SplashScreenState extends State<SplashScreen> {
     await GetIt.I.allReady();
     final sp = await SharedPreferences.getInstance();
 
-    if (!sp.containsKey('api_key')) return _navigate('/login');
+    if (!sp.containsKey('api_key')) return _navigateToLogin();
 
     final apiKey = sp.getString('api_key')!;
     const baseUrl = kIsWeb ? '' : String.fromEnvironment('base_url');
 
     final isValid = await AdminApiClient.validateApiKey(baseUrl: baseUrl, apiKey: apiKey);
-    if (!isValid) return _navigate('/login');
+    if (!isValid) return _navigateToLogin();
 
     GetIt.I.registerSingleton(await AdminApiClient.create(baseUrl: baseUrl, apiKey: apiKey));
     return _navigate('/identities');
   }
 
+  void _navigateToLogin() {
+    final redirectPath = widget.redirect != null ? Uri.encodeComponent(widget.redirect!) : '/';
+    context.go('/login?redirect=$redirectPath');
```

Comment:
no.

1. `widget.redirect` is already encoded
2. / as default?! login has a default and we don't want to override it

```suggestion
    if (widget.redirect != null) context.go('/login?redirect=${widget.redirect!}');

    context.go('/login');
```

Reply:
or even easier

```suggestion
    context.go(widget.redirect != null ? '/login?redirect=${widget.redirect!}' : '/login');
```